### ISSUE TITLE
Executing Dredd Tests on Travis CI for Play Framework Application

### ISSUE THREAD

**Issue by jesuscreyes** [apiaryio/dredd#560]
I'm trying to run Dredd tests on Travis CI for a simple Play Framework Application in Scala.

The lines in my `.travis.yml` file that call the Dredd test are configured as follows:
![screen shot 2016-07-11 at 8 40 38 am](https://cloud.githubusercontent.com/assets/9061700/16736757/3a9f110c-4743-11e6-9622-bb42a5038502.png)

And in my `dredd.yml` file I have the configuration as follows:
![screen shot 2016-07-11 at 8 42 33 am](https://cloud.githubusercontent.com/assets/9061700/16736810/72341c7a-4743-11e6-96d5-0d5ce2e3fb14.png)

I'm able to run my Dredd tests locally, but not through Travis CI. The error I get in Travis is this:
<img width="591" alt="screen shot 2016-07-11 at 8 37 30 am" src="https://cloud.githubusercontent.com/assets/9061700/16736699/02f2e0da-4743-11e6-8a20-abeab46a1eeb.png">

For some reason, the server is automatically stopped before the Dredd tests are actually executed on Travis CI.

Why is the server stopping before the Dredd tests start on Travis CI? Is there a way to resolve this issue?

**Comment by netmilk** [comment]
Hey @jesuscreyes!

I'm excited you're giving Dredd a try! I'm sorry for the inconvenience and thank you for reporting it.

Would you mind sending the `dredd.yml` again? I'd need to see it from the very first line. I think the key could lay there.

Is the backend server set to spawn to background and run as daemonized? Can you increase log level of your backend server?

I wouldn't suspect Dredd doing something wrong or unexpected in your case. The `info: Backend server exited` means your backend server process ended. If it was Dredd who tried to terminate it, it would emit either `Sending SIGTERM to the backend server` or `Killing backend server`.

**Comment by jesuscreyes** [comment]
Hi @netmilk, thanks for your help. Sorry, I forgot to clarify something. So my Dredd Test works on my local machine fine. It's when I'm running the Dredd Test using the Travis CI machines where I run into this issue.

So here's a snapshot of what I get on my local machine (I'm able to execute tests perfectly):
![screen shot 2016-07-12 at 3 01 16 pm](https://cloud.githubusercontent.com/assets/9061700/16785203/97a75ff4-4841-11e6-9de8-d1293c4b29b3.png)

Here's my whole dredd.yml:
![screen shot 2016-07-12 at 2 56 56 pm](https://cloud.githubusercontent.com/assets/9061700/16785075/e8e8e55a-4840-11e6-84fd-6bc93363ef76.png)

**Comment by w-vi** [comment]
@jesuscreyes Debugging CI issues is terrible, but it looks that for some reason your server shuts down almost straight away. Can you start the server out of dredd and give dredd a nop server start so that the server is decoupled from dredd? something along these lines:
```
before_script:
  - server_start
  - dredd --server "echo X"
  - server_stop

```

**Comment by jesuscreyes** [comment]
Hi @w-vi, I tried out what you suggested, but I encounter the same issue. This is the script I used.

```
before_script:
  - sbt run
  - dredd --server "echo X"
```

This is the result I get:
<img width="585" alt="screen shot 2016-07-13 at 11 01 05 am" src="https://cloud.githubusercontent.com/assets/9061700/16814086/2947e2ce-48e9-11e6-96ed-e7ea5dfd5b94.png">

My conclusion from this is that `sbt run` is not working correctly on the Travis CI machines. So I no longer think that this is a Dredd issue. I'll move the discussion over to the Travis people and try to find an answer there.

Thanks for your help and suggestions!
